### ISSUE TITLE
Mishandling of missing data in footprints causes a fails of the inversion

### ISSUE THREAD

**Issue by alexdanjou** [openghg/openghg_inversions#172]
l. 290 - 309 of get_data.py (see "Copy of the problematic lines" below) does not work properly : if no data are in the data range, the search error is not raised and thus the site not removed. This occurs in an error further down the road (see "Full error message" below).

This may be an openghg issue rather than openghg_inversions though. My guess is that the function _get_footprints_ checks the start and end dates but do not see if there is a missing period in between. But it is a guess as I had difficulties understanding the openghg code.

**Copy of the problematic lines :**
```
#Get footprints data
        try:
            get_fps = get_footprint(
                site=site,
                height=fp_height[i],
                domain=domain,
                model=fp_model,
                met_model=met_model[i],
                start_date=start_date,
                end_date=end_date,
                store=footprint_store,
                species=fp_species,
            )
        except SearchError:
            print(
                f"\nNo footprint data found for {site} with inlet/height {fp_height[i]}, model {fp_model}, and domain {domain}.",
                f"Check these values.\nContinuing model run without {site}.\n",
            )
            continue  # skip this site
        else:
            footprint_dict[site] = get_fps
```

**Full error message:**
```
Traceback (most recent call last):
  File "/user/home/bq24992/openghg_inversions/openghg_inversions/hbmcmc/run_hbmcmc.py", line 248, in <module>
    mcmc_function(**param)
  File "/user/home/bq24992/openghg_inversions/openghg_inversions/hbmcmc/hbmcmc.py", line 305, in fixedbasisMCMC
    ) = get_data.data_processing_surface_notracer(
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/openghg_inversions/openghg_inversions/get_data.py", line 354, in data_processing_surface_notracer
    scenario_combined = model_scenario.footprints_data_merge()
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/openghg/analyse/_scenario.py", line 1563, in footprints_data_merge
    combined_dataset = self.combine_obs_footprint(
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/openghg/analyse/_scenario.py", line 795, in combine_obs_footprint
    aligned_obs, aligned_footprint = self._align_obs_footprint(resample_to=resample_to, platform=platform)
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/openghg/analyse/_scenario.py", line 738, in _align_obs_footprint
    footprint_data = footprint_data.resample(indexer={"time": resample_period}, base=base).mean()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/xarray/core/dataset.py", line 10554, in resample
    return self._resample(
           ^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/xarray/core/common.py", line 1092, in _resample
    rgrouper = ResolvedGrouper(grouper, group, self)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 6, in __init__
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/xarray/core/groupby.py", line 440, in __post_init__
    self.group: T_Group = _resolve_group(self.obj, self.group)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/user/home/bq24992/.conda/envs/openghg_inv/lib/python3.11/site-packages/xarray/core/groupby.py", line 785, in _resolve_group
    raise ValueError(f"{newgroup.name} must not be empty")
ValueError: __resample_dim__ must not be empty

```
It can be reproduced by doing ch4 monthly inversion with site ZEP for 2018-02-01 to 2018-03-01.

**Comment by brendan-m-murphy** [comment]
I think this might be an OpenGHG issue. For flux and boundary conditions, I think we expect the search to essentially ignore the start and end dates, but this seems to be happening for footprints as well.

**Comment by alexdanjou** [comment]
Actually same as issue  #196 and solved (in openghg_inversions) by #195 .

Error message is different from #196 due to changes in openghg_inversions/openghg in between.

**Comment by alexdanjou** [comment]
Closing the issue
